apps/api/src/app/events/e2e/digest-events.e2e.ts

```diff
@@ -591,4 +591,64 @@ describe('Trigger event - Digest triggered events - /v1/events/trigger (POST)',
 
     expect(job.digest.events.length).to.equal(3);
   });
+
+  it('should create multiple digest based on diffrent digestKeys', async function () {
+    const postId = MessageRepository.createObjectId();
+    template = await session.createTemplate({
+      steps: [
+        {
+          type: ChannelTypeEnum.DIGEST,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5,
+            digestKey: 'postId',
+            type: DigestTypeEnum.REGULAR,
+          },
+        },
+        {
+          type: ChannelTypeEnum.SMS,
+          content: 'Hello world {{postId}}' as string,
+        },
+      ],
+    });
+
+    await triggerEvent({
+      customVar: 'Testing of User Name',
+      postId,
+    });
+
+    await triggerEvent({
+      customVar: 'digest',
+      postId: MessageRepository.createObjectId(),
+    });
+
+    let digests = await jobRepository.find({
+      _templateId: template._id,
+      type: ChannelTypeEnum.DIGEST,
+    });
+
+    expect(digests[0].payload.postId).not.to.equal(digests[1].payload.postId);
+    expect(digests.length).to.equal(2);
+
+    await awaitRunningJobs(2);
+
+    digests = await jobRepository.find({
+      _templateId: template._id,
+      type: ChannelTypeEnum.DIGEST,
+    });
+
+    for (const digest of digests) {
+      await workflowQueueService.work(digest);
+    }
+
+    await awaitRunningJobs(0);
+
+    const messages = await messageRepository.find({
+      _templateId: template._id,
+    });
+
+    expect(messages[0].content).to.include(digests[0].payload.postId);
+    expect(messages[1].content).to.include(digests[1].payload.postId);
```

Comment:
@davidsoderberg maybe also worth adding event count to the generated template to make sure it is created as expected?